charts/consul/templates/connect-inject-clusterrole.yaml

```diff
@@ -49,5 +49,10 @@ rules:
   - {{ template "consul.fullname" . }}-connect-inject-acl-token
   verbs:
   - get
+- apiGroups: [""]
+  resources:
+    - serviceaccounts
+  verbs:
+    - get
```

Comment:
So since we're mounting the service account secret on the volume, the reason we're getting the service account in the handler is to [get the secret name](https://github.com/hashicorp/consul-k8s/pull/1012/files/9e7920439cc3b8f2e3aae447e06ef787dab2b480#diff-9edb67e94492a07a3800e8add0f97513e33e559038ec532ced787d9af7b63f09R284) (and to do validation that the service account exists). The secret name would be something like: `web-token-bvbk5` so we can't infer the whole name from the service account name, so I think we need this perm for that

Reply:
Ah ok! that makes sense. I thought for a second that we may be able to just mount the secret but you're right we can't infer it. I guess the only question is whether we should put it behind a feature flag since we only need it for multiport? I think it's also OK to leave as is and change if someone raises an issue.

Reply:
Yeah, that sounds good, I'll go ahead and leave it then and see if it becomes an issue to allow this extra perm